compiler/ifa/num.h

```diff
@@ -79,6 +79,10 @@ class Immediate { public:
   uint32_t const_kind;
   uint32_t num_index;
   union {
+    // Unions are initalized based off the first element, so we need to have
+    // the largest thing first to make sure it is all zero initalized
+    complex128 v_complex128;
+    complex64  v_complex64;
```

Comment:
I get that this is necessary, but it's a little unfortunate that it means booleans and numerical types get intermixed (and that types both increase and decrease in size).  This doesn't need to be changed in this patch, but it's something I wanted to point out while I was thinking about it